client/src/components/Announcements.js

```diff
@@ -10,11 +15,54 @@ class AnnouncementList extends Component {
     constructor(props) {
       super(props);
       this.state = {
-        announcements : this.props.announcements,
+        announcements : [],
           id: this.props.id
       }
+      
+    }
+
+    /**
+     * Makes a GET request to either dev or prod and fetches the announcement list
+     * as soon as the component loads and updates the state
+     * TODO: Add Validation for output
+     * TODO: Once finished, remove debugging statements
+     */
+    componentDidMount() {
+        let urlToFetch = API_ANNOUNCEMENTS_PROD;
+        if (this.ReactIsInDevelomentMode()) {
+            urlToFetch = API_ANNOUNCEMENTS_DEV;
+        }
+        fetch(urlToFetch)
+        .then(response => response.json())
+        .then(data =>  this.setState({announcements:this.parseAnnouncements(data)}))
+          
     }
 
+    /**
+     * TODO: How are multiple announcements stored? in data.annoucnment or just as a string
+     * Assuming the object has an attribute announcement for now
+     * 
+     * Returns formatted data to be stored in the state
+     */
+    parseAnnouncements(data) {
+        if (data[0].default !== undefined) {
```

Comment:
The condition for this should be switched. You want to check if there is an announcements key first and if it's not there, then you should use default.